cfg/config.go

```diff
@@ -251,17 +252,30 @@ type SerialTNCConfig struct {
 
 	// Type of TNC (currently only 'kenwood').
 	Type string `json:"type"`
+
+	// (optional) Reference name to the Hamlib rig for frequency control.
+	Rig string `json:"rig"`
 }
 
-type AX25Config struct {
-	// axport to use (as defined in /etc/ax25/axports).
-	Port string `json:"port"`
+type AGWPEConfig struct {
+	// The TCP address of the TNC.
+	Addr string `json:"addr"`
 
-	// Optional beacon when listening for incoming packet-p2p connections.
-	Beacon BeaconConfig `json:"beacon"`
+	// The AGWPE "radio port" (0-3).
+	RadioPort int `json:"radio_port"`
+}
+
+type AX25Config struct {
+	Engine AX25Engine `json:"engine"`
```

Comment:
It feels slightly wonky to put the engine here, but I suppose it strikes a balance between backward-compatibility and clean config.

Reply:
Yes, I agree. Maybe it's worth revisiting. I'll try a bit more to come up with a better solution. Feel free to share if you have any ideas 🙂 It might be worth designing this without thinking about backward-compatibility, and then try to fix the compatibility with some re-write rules.

Reply:
Brainstorming a bit...

# Alternative 1 (current)

I feel we need a common struct to group common AX.25 fields that applies to all engines (beacon and rig config as well as the selected engine). The natural name for this is `ax25`, which we have used for the linux-specific implementation until now.

My main concern with this is that the field name `port` does not reflect the fact that it only applies to the linux engine. 

```json
{
  "ax25": {
    "engine": "linux",
    "port": "wl2k",
    "beacon": {"every": "600", ...},
    "rig": "tmd710"
  },
  "agwpe": {
    "addr": "localhost:8000",
    "radio_port": 0
  }
}
```

# Alternative 2

Following the same pattern as with other modem config, we could move the `port` field into a linux-specific, reserving `ax25` for common fields that applies to all engines.

Today we only have one config field that is specific for `ax25+linux`, but if we in the future get rid of the libax25 dependency (and it's system-wide axport config file) we could see the need to expand with several new fields (`tty`, `serial_baud` etc).

```json
{
  "ax25": {
    "engine": "linux",
    "beacon": {"every": "600", ...},
    "rig": "tmd710"
  },
  "ax25_linux": {
    "port": "wl2k"
  },
  "agwpe": {
    "addr": "localhost:8000",
    "radio_port": 0
  }
}
```

# Alternative 3

We could also just consider renaming the "port" field to better reflect the fact that it only applies to the linux engine. 

```json
{
  "ax25": {
    "engine": "linux",
    "linux_port": "wl2k",
    "beacon": {"every": "600", ...},
    "rig": "tmd710"
  },
  "agwpe": {
    "addr": "localhost:8000",
    "radio_port": 0
  }
}
```

# Alternative 4

We could consider using "packet" instead of "ax25" in both the scheme and config structs, which would create more distance between the scheme and the mode/implementation. This would also be more in line with the wording used in Winlink Express. We could then consider either "linux" or "linux_ax25" as the engine name.

```json
  "packet": {
    "engine": "linux_ax25",
    "beacon": {"every": "600", ...},
    "rig": "tmd710"
  },
  "linux_ax25": {
    "port": "wl2k"
  },
  "agwpe": {
    "addr": "localhost:8000",
    "radio_port": 0
  }
}

```

Reply:
Regarding Alt. 4, I think we should then rename the scheme to `packet://`, `packet+agwpe://`, `packet+linux_ax25://`....

Reply:
I'm leaning towards Alt. 2 for the first iteration of this. It's not perfect, but it makes it a lot clearer IMO.

Reply:
Pushed a new commit (alternative 2).

Reply:
Ok, this looks better to me. Thanks!

Reply:
Excellent. I'm pretty happy with it now too 🙂 Thanks for the feedback Chris! 🙇